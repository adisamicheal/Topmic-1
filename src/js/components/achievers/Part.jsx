import React from 'react';
import { string } from 'prop-types';

const Part = ({ text, textHeading, image }) => (
	<div className="part">
		<img src={image} alt="" />
		<div className="paragraph">
			<h4>{textHeading}</h4>
			<p>{text}</p>
			<div className="clear" />
		</div>
		<div className="clear" />
	</div>
);

export default Part;

Part.propTypes = {
	text: string.isRequired,
	textHeading: string.isRequired,
	image: string.isRequired
};
