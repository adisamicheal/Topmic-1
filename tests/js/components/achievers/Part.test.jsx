import React from 'react';
import Enzyme,{shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

import Part from '../../../../src/js/components/achievers/Part';

Enzyme.configure({adapter:new Adapter()});

describe('Test the Part Component',()=>{
	const wrapper=shallow(<Part text={''} textHeading={''} image={''} />)
	it('test the rendring',()=>{
		expect(wrapper).toMatchSnapshot();
	})
})