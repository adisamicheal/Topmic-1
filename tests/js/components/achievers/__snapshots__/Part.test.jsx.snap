// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Part Component test the rendring 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Part
    image=""
    text=""
    textHeading=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          src=""
        />,
        <div
          className="paragraph"
        >
          <h4>
            
          </h4>
          <p>
            
          </p>
          <div
            className="clear"
          />
        </div>,
        <div
          className="clear"
        />,
      ],
      "className": "part",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              
            </h4>,
            <p>
              
            </p>,
            <div
              className="clear"
            />,
          ],
          "className": "paragraph",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "clear",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "clear",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            src=""
          />,
          <div
            className="paragraph"
          >
            <h4>
              
            </h4>
            <p>
              
            </p>
            <div
              className="clear"
            />
          </div>,
          <div
            className="clear"
          />,
        ],
        "className": "part",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                
              </h4>,
              <p>
                
              </p>,
              <div
                className="clear"
              />,
            ],
            "className": "paragraph",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "clear",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "clear",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
